src/sentry/integrations/slack/views/unlink_team.py

```diff
@@ -58,25 +57,26 @@ def handle(self, request: Request, signed_params: str) -> Response:
         channel_name = params["channel_name"]
         channel_id = params["channel_id"]
 
-        try:
-            external_team = ExternalActor.objects.get(
-                organization=organization,
-                integration=integration,
-                provider=ExternalProviders.SLACK.value,
-                external_name=channel_name,
-                external_id=channel_id,
-            )
-        except IntegrityError as e:
-            logger.error("slack.team.unlink.integrity-error", extra=e)
+        external_teams = ExternalActor.objects.filter(
+            organization=organization,
+            integration=integration,
+            provider=ExternalProviders.SLACK.value,
+            external_name=channel_name,
+            external_id=channel_id,
+        )
+        if len(external_teams) == 0:
+            logger.error("slack.team.unlink.no_external_teams")
```

Comment:
I changed the logger message from "slack.team.unlink.integrity-error" to this since it's not an integrity error really